core/probe.cpp

```diff
@@ -396,6 +398,11 @@ void Probe::delayedInit()
         showInProcessUi();
 }
 
+void Probe::shutdown()
+{
+    delete this;
```

Comment:
Reading http://stackoverflow.com/questions/550189/is-it-safe-to-delete-this I think it's fine to have it this way though